src/tests/test_process.py

```diff
@@ -32,10 +54,27 @@ class TestProcessProto(unittest.TestCase):
     @classmethod
     def setUpClass(cls):
         """Setup environment and start pybitmessage"""
-        cls.home = os.environ['BITMESSAGE_HOME'] = tempfile.gettempdir()
+        cls.home = set_appdata_folder(tempfile.gettempdir())
+        cls._cleanup_files()
         put_signal_file(cls.home, 'unittest.lock')
+        starttime = time.time()
         subprocess.call(cls._process_cmd)  # nosec
-        time.sleep(5)
+        timeout = starttime + 30.0
+        while time.time() <= timeout:
+            if os.path.exists(os.path.join(cls.home,
+                                           'singleton.lock')):
+                try:
+                    pstat = os.stat(os.path.join(cls.home, 'singleton.lock'))
+                    if starttime <= pstat.st_mtime and pstat.st_size > 0:
+                        break
+                except OSError:
+                    break
```

Comment:
As I can see `OSError` raised when `os.path.exists(os.path.join(cls.home, 'singleton.lock')) is False`.

Reply:
Oh, now I get it.